hack/generator/pkg/codegen/pipeline_mark_storage_version.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+	"sort"
+
+	"github.com/Azure/k8s-infra/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+)
+
+// exportPackages creates a PipelineStage to export our generated code as a set of packages
+func markStorageVersion() PipelineStage {
+	return MakePipelineStage(
+		"markStorageVersion",
+		"Marking the latest version of type as the storage version",
+		func(ctx context.Context, types astmodel.Types) (astmodel.Types, error) {
+			updatedDefs, err := MarkLatestResourceVersionsForStorage(types)
+			if err != nil {
+				return nil, errors.Wrapf(err, "unable to mark latest resource versions for as storage versions")
+			}
+
+			return updatedDefs, nil
+		})
+}
+
+// MarkLatestResourceVersionsForStorage marks the latest version of each resource as the storage version
+func MarkLatestResourceVersionsForStorage(types astmodel.Types) (astmodel.Types, error) {
+
+	result := make(astmodel.Types)
+	resourceLookup, err := groupResourcesByVersion(types)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, def := range types {
+		// see if it is a resource
+		if resourceType, ok := def.Type().(*astmodel.ResourceType); ok {
+
+			unversionedName, err := getUnversionedName(def.Name())
+			if err != nil {
+				// should never happen as all resources have versioned names
+				return nil, err
+			}
+
+			allVersionsOfResource := resourceLookup[unversionedName]
+			latestVersionOfResource := allVersionsOfResource[len(allVersionsOfResource)-1]
```

Comment:
Can't just take the latest version, we want to avoid picking preview releases for storage as they're too volatile.

See https://github.com/Azure/k8s-infra/blob/master/hack/generator/spec/case-study-rolling-storage-versions.md#version-2014-04-04-preview---schema-change

Reply:
See comment about about not changing the logic we had, just refactoring -- same questions apply here.